HW06.Rmd

```diff
@@ -1,92 +1,147 @@
+
+library(tidyverse)
+library(magrittr)
+library(dplyr)
+
+# list.files : produces a character vector of the names of files in the named directory
+H3K27M_names <- list.files("./RNA_Seq_processed/H3K27M", full.names=TRUE)
+WT_names <- list.files("./RNA_Seq_processed/WT", full.names=TRUE)
+
+#create read_in function to read in only TPM
+read_in <- function(x){
+  read_tsv(x) %>%
+  select(contains("TPM"))
+}
+
+#read in the data using read_in defined function above in the format of a data.frame
+H3K27M <- data.frame(lapply(H3K27M_names, read_in))
+WT <- data.frame(lapply(WT_names, read_in))
+
+#get name col
+name_col <- read_tsv(H3K27M_names[1]) %>%
+  select(Name) %>%
+  rename(gene_id = Name)
+
+#combine the data (list with all files)
+all_data <- c(H3K27M, WT) 
+
+#create data frame GBM.transcripts as indicated below
+GBM.transcripts <- bind_cols(name_col, all_data)
+
+
+```
+
+Now, install the packages commented below (if needed), and then use this code to map the transcript IDs to gene symbols. To use this code, you need a dataframe called GBM.transcripts that has the first column "gene_id" that contains the transcript ids (e.g. ENST00000456328.2) and the remaining columns contain the TPM data. So long as the first column contains the "gene_id" column as mentioned above, this *should* run. 
+
+```{r map gene.ids to gene.symbols using ensembldb}
+
+#install.packages("BiocManager")
+#BiocManager::install("ensembldb")
+#BiocManager::install("EnsDb.Hsapiens.v75")
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+
```

Comment:
I was wondering what other gene(s) you would be interested in but I realized you might have forgotten to add it!